machineconfiguration/v1alpha1/types_machineconfignode.go

```diff
@@ -107,6 +111,36 @@ type MachineConfigNodeStatus struct {
 	// This desired machine config has been compared to the current machine config and has been validated by the machine config operator as one that is valid and that exists.
 	// +kubebuilder:validation:Required
 	ConfigVersion MachineConfigNodeStatusMachineConfigVersion `json:"configVersion"`
+	// pinnedImageSetVersion describes the current and desired pinned image sets for this node.
+	// The current version is the generation of the pinned image set that has most recently been successfully pulled and pinned on this node.
+	// The desired version is the generation of the pinned image set that is targeted to be pulled and pinned on this node.
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	PinnedImageSetVersion []MachineConfigNodeStatusPinnedImageSetVersion `json:"pinnedImageSetVersion,omitempty"`
+}
+
+type MachineConfigNodeStatusPinnedImageSetVersion struct {
+	// name is the name of the pinned image set.
+	// Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123)
+	// It may consist of only alphanumeric characters, hyphens (-) and periods (.)
+	// and must be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength:=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// currentGeneration is the generation of the pinned image set that has most recently been successfully pulled and pinned on this node.
+	// +optional
+	CurrentGeneration int32 `json:"currentGeneration"`
+	// desiredGeneration version is the generation of the pinned image set that is targeted to be pulled and pinned on this node.
+	// +optional
+	DesiredGeneration int32 `json:"desiredGeneration,omitempty"`
```

Comment:
positive only.

Reply:
This should never be smaller than currentgeneration or lastFailedGeneration,  right?